File: src/main/Java/boj_step6/boj2444.java
import java.io.*;

public class boj2444 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        int i = 1;
        while (i <= n) {
            int j = 1;
            int k = n - i;
            while (k-- > 0) writer.append(" ");
            while (j++ <= i) writer.append("*");
            j = 2;
            while (j++ <= i) writer.append("*");
            writer.newLine();
            i++;
        }
        i = n;
        while (i > 0) {
            int j = 1;
            int k = n - i + 1;
            while (k-- > 0) writer.append(" ");
            while (j++ < i) writer.append("*");
            j = 2;
            while (j++ < i) writer.append("*");
            writer.newLine();
            i--;
        }
        writer.close();
    }
}

File: src/main/kotlin/boj2444.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine()!!.toInt()
    for (i in 1..n) {
        for (j in n downTo i + 1) w.write(" ")
        for (j in 1..i) w.write("*")
        for (j in 2..i) w.write("*")
        w.write("\n")
    }
    for (i in n downTo 1) {
        for (j in n downTo i) w.write(" ")
        for (j in 1..i - 1) w.write("*")
        for (j in 2..i - 1) w.write("*")
        w.write("\n")
    }
    w.close()
}